consensus/state.go

```diff
@@ -756,17 +756,32 @@ func (cs *ConsensusState) handleTimeout(ti timeoutInfo, rs cstypes.RoundState) {
 func (cs *ConsensusState) handleTxsAvailable() {
 	cs.mtx.Lock()
 	defer cs.mtx.Unlock()
-	// we only need to do this for round 0
-	cs.enterNewRound(cs.Height, 0)
-	cs.enterPropose(cs.Height, 0)
+
```

Comment:
First thing we should check is if we're in the timeout commit period.

If so, and this stuff is true, go then go to the next round.

If so, and this stuff is not true, then schedule the round timeout.

Otherwise, go to the round.